core/controller/src/main/scala/whisk/core/controller/Actions.scala

```diff
@@ -490,21 +495,35 @@ trait WhiskActionsApi extends WhiskCollectionAPI {
      * If this mechanism fails to produce an answer quickly, the future will fail and we back off into using
      * a database operation to obtain the canonical answer.
      */
-    private def pollLocalForResult(
-        docid: DocInfo,
+    private def waitForActivationResponse(
+        namespace: Namespace,
         activationId: ActivationId,
-        promise: Promise[Option[WhiskActivation]])(
-            implicit transid: TransactionId): Unit = {
-        queryActivationResponse(activationId, transid) map {
-            activation => promise.trySuccess { Some(activation) }
-        } onFailure {
+        timeout: FiniteDuration)(
+            implicit transid: TransactionId): (ActivationId, Future[WhiskActivation]) = {
```

Comment:
We need to return the activationId again, because of the unitTests.
In the test an activation is created in the db with an predefined activationId. `queryActivationResponse` returns this predefined ActivationId and to let the test pass, the action has to look in the db for the activationId, that has been written by the test.
Before this change, `performLoadBalancerRequest` had this functunality.